samcli/local/apigw/authorizers/lambda_authorizer.py

```diff
@@ -50,18 +66,17 @@ def is_valid(self, **kwargs) -> bool:
 
         Returns
         -------
-        bool
-            True if the header is present and not empty
+        Any
+            The header if it is found, otherwise None
         """
         headers = kwargs.get("headers", {})
-
-        return bool(headers.get(self.identity_source, None))
+        return headers.get(self.identity_source, None)
 
 
 class QueryIdentitySource(IdentitySource):
-    def is_valid(self, **kwargs) -> bool:
+    def find_identity_value(self, **kwargs) -> Any:
```

Comment:
Can we annotate the return type with `Optional[Something]` instead of `Any`?
where `Something` is the type